packages/core/src/messageformat.ts

```diff
@@ -61,12 +86,18 @@ export interface MessageFormatOptions<
    */
   customFormatters?: {
     [key: string]:
-      | CustomFormatter
+      | DefaultCustomFormatter
       | {
-          formatter: CustomFormatter;
+          formatter: DefaultCustomFormatter;
           arg?: 'string' | 'raw' | 'options';
           id?: string;
           module?: string;
+        }
+      | {
+          formatter: LocaleModuleCustomFormatter;
+          arg?: 'string' | 'raw' | 'options';
+          id?: string;
+          module?: (_: { locale: string }) => string;
```

Comment:
Why does this expect the `locale` to be wrapped within an object? Wouldn't it be simpler to use `(locale: string) => string`?

Reply:
I had it that way originally, but i changed it to the object form so it could be extended without being a breaking change

Reply:
Locale is good to keep as a separate argument. For example, that's the way it is in the MessageFormat constructor. Adding more arguments can be done later.

Reply:
sure, i can get on board with that. refactored in ee31644